lib/mapper.js

```diff
@@ -53,15 +53,20 @@ exports.identify = function(identify){
  */
 
 exports.track = function(track){
+  var customer_properties = {};
+  var userId = track.userId();
+  var email = track.email();
+
+  // at least one of userId or email is required by Klaviyo
+  if (userId) customer_properties.$id = userId;
+  if (email) customer_properties.$email = email;
+
   return {
     token: this.settings.apiKey,
     event: track.event(),
     properties: properties(track),
     time: time(track.timestamp()),
-    customer_properties: {
-      $id: track.userId() || track.sessionId(),
```

Comment:
Passing a $id to Klaviyo overrides their unique-email-address enforcement so many accounts are being created tied to a single email address when we fallback to anonymousId